### PR TITLE
remarkable-linux-client: init at unstable-2017-10-27

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/office/remarkable-linux-client/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    cp reMarkable $out/bin
+    cp libpdfium.so.1 $out/lib
+
+    patchelf --set-interpreter \
+      "${glibc}/lib/ld-linux-x86-64.so.2" \
+      "$out/bin/reMarkable"
+
+    patchelf \
+      --set-rpath \
+      "$out/lib:${libsForQt512.karchive}/lib:${qtdeclarative}/lib:${qtsvg}/lib:${qtbase.out}/lib:${qtwebsockets}/lib:${libGLU}/lib:${stdenv.cc.cc.lib}/lib:${libGL}/lib" \
```
Comment (r-burns) [inline pkgs/applications/office/remarkable-linux-client/default.nix:29]: Consider `lib.makeSearchPath`

Comment (flokli) [reply pkgs/applications/office/remarkable-linux-client/default.nix:29]: does `autoPatchelfHook` not work on this one?

File: pkgs/applications/office/remarkable-linux-client/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  };
+
+  buildInputs = [ stdenv.cc.cc.lib libGLU libGL libsForQt512.karchive
+                  qtbase qtdeclarative qtsvg qtbase qtwebsockets
+                  qtquickcontrols qtquickcontrols2 qtgraphicaleffects ];
+
+  installPhase = ''
+    mkdir -p $out/{bin,lib}
+    cp reMarkable $out/bin
+    cp libpdfium.so.1 $out/lib
```
Comment (SuperSandro2000) [inline pkgs/applications/office/remarkable-linux-client/default.nix:21]: Do we need a symlink from libpdfium.so to libpdfium.so.1?

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107450` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>remarkable-linux-client</li>
  </ul>
</details>

File: pkgs/applications/office/remarkable-linux-client/default.nix (hunk 1 above)
Comment (siraben) [reply pkgs/applications/office/remarkable-linux-client/default.nix:29]: I'll try using that.

**Comment by divanorama** [comment]
Tried on Gentoo amd64 with sandbox=true
- build succeeds :heavy_check_mark: 
- app connects to reMarkable cloud and sees my pdf files :heavy_check_mark: 
Great to see this app in nixpkgs! One small quirk is that I had to use nixGL on Gentoo, app seems to need openGL to start.

Note: I've also recently added cli client rmapi to my nur repo https://github.com/divanorama/nur-packages/blob/master/pkgs/rmapi/default.nix maybe can be moved to nixpkgs too

File: pkgs/applications/office/remarkable-linux-client/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      "$out/bin/reMarkable"
+
+    patchelf \
+      --set-rpath \
+      "$out/lib:${libsForQt512.karchive}/lib:${qtdeclarative}/lib:${qtsvg}/lib:${qtbase.out}/lib:${qtwebsockets}/lib:${libGLU}/lib:${stdenv.cc.cc.lib}/lib:${libGL}/lib" \
+      "$out/bin/reMarkable"
+  '';
+  meta = with stdenv.lib; {
+    description = "Linux client for interacting with reMarkable cloud";
+    homepage = "https://remarkable.engineering";
```
Comment (divanorama) [inline pkgs/applications/office/remarkable-linux-client/default.nix:34]: maybe change to https://remarkablewiki.com/tips/client
https://remarkable.engineering is almost a stub that can only take you to 2 places
- https://remarkable.com/ tablet home page
- https://github.com/reMarkable org github (note that client app code doesn't seem to be there)

File: pkgs/applications/office/remarkable-linux-client/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    license = licenses.unfree; # no source or license in tarball
```
Comment (divanorama) [inline pkgs/applications/office/remarkable-linux-client/default.nix:35]: In the about menu there's a "Legal information" button which shows licenses of opensource libs used, though still not clearly saying which license is used for the app itself
image

Comment (siraben) [reply pkgs/applications/office/remarkable-linux-client/default.nix:35]: I think this license notice is out of date.

**Comment by NickHu** [comment]
@divanorama rmapi is already in nixpkgs

**Comment by siraben** [comment]
Given that rM has stopped maintaining this as an official client, I may close this in favor of other open source solutions.

File: pkgs/applications/office/remarkable-linux-client/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  version = "unstable-2017-10-27";
+
+  src = fetchurl {
+    url = "https://remarkable.engineering/remarkable-linux-client-0.0.5-16-1408-g7eca2b66.tgz";
+    sha256 = "1305scjyi4b1wh4vr8ccszz11dvgwyka9hivyzv5j8ynqsnij58s";
+  };
+
+  buildInputs = [ stdenv.cc.cc.lib libGLU libGL libsForQt512.karchive
+                  qtbase qtdeclarative qtsvg qtbase qtwebsockets
+                  qtquickcontrols qtquickcontrols2 qtgraphicaleffects ];
```
Comment (SuperSandro2000) [inline pkgs/applications/office/remarkable-linux-client/default.nix:16]:
```suggestion
    qtbase qtdeclarative qtsvg qtbase qtwebsockets
    qtquickcontrols qtquickcontrols2 qtgraphicaleffects ];
```

**Comment by flokli** [comment]
> Given that rM has stopped maintaining this as an official client, I may close this in favor of other open source solutions.

Do you still want to add (and maintain) this as a package? Are there better, more maintained open source variants out there?

**Comment by siraben** [comment]
Closing because it would add an unmaintained, unfree package to Nixpkgs. Users should use [rmapi](https://github.com/juruen/rmapi) instead.

**Pull request closed without merging**